### PR TITLE
Add workflow to run tests with different TensorFlow versions

### PR DESCRIPTION
**Pull request by oliverholworthy**
Add workflow to run tests with different TensorFlow versions. To provide increased confidence when making changes that we support mulitple TensorFlow versions.

### CODE REVIEW

File: tests/unit/dataloader/test_tf_dataloader.py [hunk 1]
```diff
@@ -20,18 +20,22 @@
 import time
 import timeit
 
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.metrics import roc_auc_score
```
Comment (benfred) [inline tests/unit/dataloader/test_tf_dataloader.py:26]: thanks for fixing the import order here!

I do wonder how the old version got through the existing isort check though -

File: tests/unit/dataloader/test_tf_dataloader.py [hunk 2]
```diff
@@ -20,18 +20,22 @@
...
+
 from merlin.core.dispatch import HAS_GPU, make_df
 from merlin.io import Dataset
 from merlin.schema import Tags
 
+pytestmark = pytest.mark.tensorflow
```
Comment (benfred) [inline tests/unit/dataloader/test_tf_dataloader.py:32]: I like adding this ability to mark the tests based on framework! .Can we do the same thing for the JAX/PT tests too - so that we're consistent?

**Review by benfred — COMMENTED** [review]
Thanks for adding in testing different versions of TF! I think this will be really helpful

Some comments:
* We are already testing out TF as part of the 'cpu-ci' GHA workflow - which seems like will be redundant after this change. Should we remove the TF tests from that workflow? Or maybe split into workflows for the different frameworks entirely and remove the cpu-ci workflow entirely?
* Can we also test out different versions of PT in a similar way?

File: tests/unit/dataloader/test_tf_dataloader.py (hunk 1 above)
Comment (oliverholworthy) [reply tests/unit/dataloader/test_tf_dataloader.py:26]: It looks like this might be an edge case in isort. It doesn't seem to spot any imports in the wrong order after the try/except.

File: tests/unit/dataloader/test_tf_dataloader.py (hunk 2 above)
Comment (oliverholworthy) [reply tests/unit/dataloader/test_tf_dataloader.py:32]: This came from the pattern used in Merlin Models for marking tests with the names of the frameworks. I've added one to each framework test file.

**Comment by oliverholworthy** [comment]
There are some tests that are not specific to frameworks, so at least  some of the tests in cpu-ci still need to run. And since the test suite is pretty fast, haven't excluded any tests to run there. So that we have one runner that tests the whole thing together.

**Comment by oliverholworthy** [comment]
rerun tests

**Review by benfred — APPROVED** [review]
looks great!
